app/models/tool_proxy.rb

```diff
@@ -0,0 +1,135 @@
+class ToolProxy < ActiveRecord::Base
+  validates :guid, :shared_secret, :tcp_url, :base_url, :tp_half_shared_secret, presence: true
+  has_many :assignments
+
+  TOOL_PROXY_FORMAT = 'application/vnd.ims.lti.v2.toolproxy+json'.freeze
+  ENABLED_CAPABILITY = %w(Security.splitSecret).freeze
+  REQUIRED_CAPABILITIES = %w(Canvas.placements.similarityDetection).freeze
+
+  # to_json
+  #
+  # Returns a tool proxy as a hash
+  def as_json(*)
+    tool_proxy_hash = super()
+    tool_proxy_hash.merge(
+      '@context' => 'http://purl.imsglobal.org/ctx/lti/v2/ToolProxy',
+      lti_version: 'LTI-2p0', # LTI-2p0 should be used for all LTI 2.x tools
+      tool_consumer_profile: tcp_url,
+      tool_profile: tool_profile,
+      security_contract: security_contract,
+      enabled_capability: ENABLED_CAPABILITY # (Section 5.3)
+    )
+  end
+
+  # tp_half_shared_secret
+  #
+  # Generates a 128 hexadecimal character string for the Tool Provider's
+  # half of the shared secret (See section 5.6).
+  def tp_half_shared_secret
+    @_tp_half_shared_secret ||= SecureRandom.hex(64)
```

Comment:
I did not see any spot where the gem does this, I'll look again just in case though. I may have looked at the wrong branch.